qiita_db/software.py

```diff
@@ -537,29 +597,187 @@ def command(self):
         qiita_db.software.Command
             The command that this parameter set belongs to
         """
-        return self._command
+        with qdb.sql_connection.TRN:
+            sql = """SELECT command_id
+                     FROM qiita.default_parameter_set
+                     WHERE default_parameter_set_id = %s"""
+            qdb.sql_connection.TRN.add(sql, [self.id])
+            return Command(qdb.sql_connection.TRN.execute_fetchlast())
 
-    def to_str(self):
-        """Generates a string with the parameter values
+
+class Parameters(object):
+    """Represents an instance of parameters used to process an artifact
+
+    This class should never be isntantiated using the __init__ method. The
```

Comment:
Thanks for the link to the `inspect` usage - we could do it using that route. For others that want to know more about it, here is the link that @antgonza sent me http://blog.lerner.co.il/making-init-methods-magical-with-autoinit/ and here is the documentation https://docs.python.org/2/library/inspect.html